### DISCUSSION TITLE
Getting empty pageProps object from SSG with catch-all route until external data is fetched

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#21335] [Help]
Whenever using:

```
pages/some-path/[[...slug]].tsx
```

I need to fetch data from an external source in getStaticProps. Before this data is fetched, [[...slug]] returns an empty object as pageProps.

In my _app I create a loader until this object is not empty anymore. It all works fine, but If I look at my page source, I see the loader in the html, and not the actual content.

Anyone know how I can fix this?

**Comment by SahasSaurav** [comment]
use useSwr or reactQuery to detect loading state in next 
can u share code or repo link
